### PR TITLE
Explain "uhandled error" in `reportFailure` scaladoc

### PR DESCRIPTION
**Pull request by kamilkloch**
h/t @armanbilge #3768

### CODE REVIEW

File: core/jvm/src/main/scala/cats/effect/IOApp.scala [hunk 1]
```diff
@@ -233,11 +233,25 @@ trait IOApp {
       )
 
   /**
-   * Configures the action to perform when unhandled errors are caught by the runtime. By
-   * default, this simply delegates to [[cats.effect.std.Console!.printStackTrace]]. It is safe
-   * to perform any `IO` action within this handler; it will not block the progress of the
-   * runtime. With that said, some care should be taken to avoid raising unhandled errors as a
-   * result of handling unhandled errors, since that will result in the obvious chaos.
+   * Configures the action to perform when unhandled errors are caught by the runtime. An
+   * unhandled error is an error that is raised on a Fiber that nobody is joining.
```
Comment (durban) [inline core/jvm/src/main/scala/cats/effect/IOApp.scala]: Suggested change: "An unhandled error is an error that is raised (and not handled) on a Fiber...".

Comment (kamilkloch) [reply core/jvm/src/main/scala/cats/effect/IOApp.scala]: Thank you, done!

**Review by durban — DISMISSED** [review]

**Comment by armanbilge** [comment]
We can retarget this PR to `series/3.5.x` branch so that the docs improvements are released sooner.

**Comment by kamilkloch** [comment]
> We can retarget this PR to `series/3.5.x` branch so that the docs improvements are released sooner.

I guess I need to re-do the PR? Tried to change the merge branch to 3.5.x, ended up with hunderds of commits.

**Review by durban — APPROVED** [review]

**Pull request merged**